ui/components/app/multichain-transaction-details-modal/multichain-transaction-details-modal.tsx

```diff
@@ -0,0 +1,365 @@
+import React, { useContext } from 'react';
+import { capitalize } from 'lodash';
+import {
+  Display,
+  FlexDirection,
+  AlignItems,
+  JustifyContent,
+  TextVariant,
+  IconColor,
+  FontWeight,
+  TextColor,
+  TextAlign,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  Modal,
+  Box,
+  Text,
+  ModalFooter,
+  Button,
+  IconName,
+  ButtonVariant,
+  Icon,
+  IconSize,
+  ButtonSize,
+  ButtonLink,
+  ButtonLinkSize,
+} from '../../component-library';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import { openBlockExplorer } from '../../multichain/menu-items/view-explorer-menu-item';
+import { ConfirmInfoRowDivider as Divider } from '../confirm/info/row';
+import { MultichainNetwork } from '../../../selectors/multichain';
+import {
+  formatDateWithYearContext,
+  shortenAddress,
+} from '../../../helpers/utils/util';
+
+type Asset = {
+  fungible: boolean;
+  type: string;
+  unit: string;
+  amount: string;
+};
+
+type TransactionFrom = {
+  address: string;
+  asset: Asset;
+};
+
+type TransactionTo = {
+  address: string;
+  asset: Asset;
+};
+
+type Transaction = {
+  type: string;
+  status: string;
+  timestamp?: number;
+  id: string;
+  from: TransactionFrom[];
+  to: TransactionTo[];
+  fees: {
+    type: string;
+    asset: Asset;
+  }[];
+};
```

Comment:
We could re-use the type coming from `@metamask/keyring-api` instead?
- https://github.com/MetaMask/accounts/blob/%40metamask/keyring-api%4013.0.0/packages/keyring-api/src/api/transaction.ts#L284